apps/enc.c

```diff
@@ -601,22 +601,22 @@ static void show_ciphers(const OBJ_NAME *name, void *arg)
         BIO_printf(dec->bio, " ");
 }
 
-static int set_hex(char *in, unsigned char *out, int size)
+static int set_hex(const char *in, unsigned char *out, int size)
 {
     int i, n;
     unsigned char j;
 
+    i = size * 2;
     n = strlen(in);
-    if (n > (size * 2)) {
+    if (n > i) {
         BIO_printf(bio_err, "hex string is too long\n");
-        return 0;
-    }
+        n = i; /* ignore exceeding part */
+    } else if (n < i)
+        BIO_printf(bio_err, "hex string is too short\n");
```

Comment:
Style NIT: this block should be surrounded by '{' & '}' because the previous part of the if is.

Reply:
You also need a better message explaining what is going to occur.  I.e. _hex string is too short, padding with zero bytes to length_.

Reply:
Agree again. Will be changed, thanks